src/org/labkey/test/tests/SampleTypeLookupDisplayColumnTest.java

```diff
@@ -0,0 +1,184 @@
+
+        new PortalHelper(init.getDriver()).addBodyWebPart("Sample Types");
+    }
+
+    private void doSetup()
+    {
+        _containerHelper.createProject(getProjectName(), null);
+    }
+
+    @Before
+    public void preTest()
+    {
+        goToProjectHome();
+    }
+
+    @Test
+    public void testLookupByIntColumn()
+    {
+        // specify fkDisplayColumnName in metadata in the list
+        setFKDisplayColumnName("ingredient", "intColumn");
+        DataRegionTable sampleTypeList =  DataRegionTable.DataRegion(getDriver()).withName("Material").waitFor();
+        sampleTypeList.clickInsertNewRow();
+
+        String comment = "inserted with lookup by ingredient.intcolumn";
+        setFieldValues("intSample", comment, "3.5", "2");
+
+        sampleTypeList =  DataRegionTable.DataRegion(getDriver()).withName("Material").waitFor();
+        var row = sampleTypeList.getRowDataAsMap("Comment", comment);
+        assertEquals("Expect Ingredient display to respect fKDisplayCol setting of intColumn",
+                "2", row.get("Ingredient"));
+    }
+
+    @Test
+    public void testLookupByTextColumn()
+    {
+        // specify fkDisplayColumnName in metadata in the list
+        setFKDisplayColumnName("ingredient", "titleColumn");
+        DataRegionTable sampleTypeList =  DataRegionTable.DataRegion(getDriver()).withName("Material").waitFor();
+        sampleTypeList.clickInsertNewRow();
+
+        String comment = "inserted with lookup by ingredient.titleColumn";
+        setFieldValues("nameSample", comment, "3.5", "sodium hexafluoride");
+
+        sampleTypeList =  DataRegionTable.DataRegion(getDriver()).withName("Material").waitFor();
+        var row = sampleTypeList.getRowDataAsMap("Comment", comment);
+        assertEquals("Expect Ingredient display to respect fKDisplayCol setting of titleColumn",
+                "sodium hexafluoride", row.get("Ingredient"));
+
+        sampleTypeList.clickEditRow(sampleTypeList.getRowIndex("Name", "nameSample"))
+                .setField("comment", "Switching to Helium to prove edit respects fkDisplay")
+                .setField("ingredient", "helium")
+                .submit();
+
+        var updatedRow = sampleTypeList.getRowDataAsMap("Name", "nameSample");
+        assertEquals("Expect updated ingredient displayed as titleColumn",
+                "helium", updatedRow.get("Ingredient"));
+
+    }
+
+    // it would be nice if our
+    private void setFieldValues(String name, String comment, String amount, String ingredient)
+    {
+        if (name != null)   // for update, name field is disabled
+            setFormElement(Locator.name("quf_Name"), name);
+        setFormElement(Locator.name("quf_comment"), comment);
+        setFormElement(Locator.name("quf_amount"), amount);
+        setFormElement(Locator.name("quf_ingredient"), ingredient);
+        clickAndWait(Locator.lkButton("Submit"));
+    }
```

Comment:
`UpdateQueryRowPage` (returned by `DataRegionTable.clickEditRow`) might work for inserting new rows also. If you try it and it works, consider updating `DataRegionTable. clickInsertNewRow` to return one.
```suggestion
    private void setFieldValues(String name, String comment, String amount, String ingredient)
    {
        var insertPage = new UpdateQueryRowPage(this);
        if (name != null)   // for update, name field is disabled
            insertPage.setField("Name", name);
        insertPage.setField("comment", comment);
        insertPage.setField("amount", amount);
        insertPage.setField("ingredient", ingredient);
        insertPage.submit();
    }
```

Reply:
I considered creating a page class for an `InsertQueryRowPage` and updating `DataRegionTable.clickInsertNewRow()` to return that, but I figured that would be seen as too much feature creep.  Thanks for suggesting!
It turns out that UpdateQueryRowPage works just fine for the insert scenario!